satsense/image.py

```diff
@@ -276,32 +272,48 @@ def save(self, filename_prefix='', extension='nc'):
                 self._save_as_netcdf(data, filename, window)
             elif extension.lower() == 'tif':
                 self._save_as_tif(data, filename, window)
+        return filenames
 
     def _save_as_netcdf(self, data, filename, window):
         """Save feature vector as NetCDF file."""
+        if len(data.shape) == 2:
+            height, width = data.shape
+            value_length = 1
+            data = data[:, :, np.newaxis]
+        elif len(data.shape) == 3:
+            height, width, value_length = data.shape
+
         with Dataset(filename, 'w') as dataset:
+            # Metadata
+            dataset.history = 'Created ' + time.ctime(time.time())
+            dataset.source = 'Satsense version ' + __version__
+            dataset.description = (
+                'Satsense extracted values for feature: ' + self.feature.name)
+            dataset.title = self.feature.name
             dataset.window = window
             dataset.arguments = repr(self.feature.kwargs)
-            dataset.createDimension('size', data.shape[-1])
-            if len(data.shape) == 2:
-                dataset.createDimension('length', data.shape[0])
-                variable = dataset.createVariable(
-                    self.feature.name, 'f4', dimensions=('length', 'size'))
-            elif len(data.shape) == 3:
-                width, height = data.shape[:2]
-                dataset.createDimension('width', width)
-                dataset.createDimension('height', height)
-                variable = dataset.createVariable(
-                    self.feature.name,
-                    'f4',
-                    dimensions=('width', 'height', 'size'))
-            variable[:] = data
+            dataset.Conventions = 'CF-1.5'
+
+            dimensions = self.add_lat_lon_dimensions(dataset, height, width)
+
+            # Actually add the values
+            dataset.createDimension('length', value_length)
+            variable = dataset.createVariable(
+                self.feature.name, 'f4', dimensions=('length', *dimensions))
+            variable.grid_mapping = 'spatial_ref'
+            variable.long_name = self.feature.name
+
+            transposed = np.moveaxis(data, source=2, destination=0)
+            variable[:] = transposed
 
     def _save_as_tif(self, data, filename, window):
         """Save feature array as GeoTIFF file."""
-        width, height, size = data.shape
-        data = np.ma.filled(data)
+        height, width, size = data.shape
         fill_value = data.fill_value if np.ma.is_masked(data) else None
+
+        if np.ma.is_masked(data):
+            msk = (~data.mask * 255).astype('uint8')
+        # data = np.ma.filled(data)
         # This is probably wrong
```

Comment:
Did you check if the moveaxis does the right thing? If so, can you remove this comment?

Reply:
What I mean by "the right thing" is that it does not shuffle the feature vectors from different coordinates.

Reply:
According to the test_save_load_roundtrip_nc test it works  as intended